libs/langchain/langchain/callbacks/tracers/evaluation.py

```diff
@@ -116,8 +116,28 @@ def _evaluate_in_project(self, run: Run, evaluator: langsmith.RunEvaluator) -> N
                 eval_result = self.client.evaluate_run(run, evaluator)
             with manager.tracing_v2_enabled(
                 project_name=self.project_name, tags=["eval"], client=self.client
-            ):
-                eval_result = self.client.evaluate_run(run, evaluator)
+            ) as cb:
+                reference_example = (
```

Comment:
code foot print is a bit larger than I would've expected to propgate through a run id

Reply:
Ya it's mainly because cyclic deps and langchain's explicit callback system